### PR TITLE
refactor: Split `test_utils.rs` to avoid having code with `test_features` and without in one file.

### PR DESCRIPTION
**Pull request by pmnoxx**
`test_utils.rs` currently has code code that's under `test_features` feature. It's better to split the code into two parts.

### CODE REVIEW

**Review by chefsale — APPROVED** [review]

File: chain/network/src/test_utils_with_test_features.rs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+};
+use actix::actors::mocker::Mocker;
+use actix::{Actor, Addr};
+use near_primitives::block::GenesisId;
+use near_primitives::network::PeerId;
+use near_store::test_utils::create_test_store;
+use near_store::Store;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+
```
Comment (mm-near) [inline chain/network/src/test_utils_with_test_features.rs:16]: and then this file would become 'test_utils"

File: chain/network/src/test_utils.rs [hunk 2]
```diff
@@ -306,89 +276,3 @@ impl MockPeerManagerAdapter {
         self.requests.write().unwrap().pop_front()
```
[the remark this answers is not in the document]
Comment (pmnoxx) [reply chain/network/src/test_utils.rs:276]: @mm-near That code is used by tests, just by tests that aren't under feature flag,

**Comment by pmnoxx** [comment]
@mm-near Can you take a look?

**Comment by pmnoxx** [comment]
@mm-near Can you take a look?

File: chain/network/src/test_utils.rs [hunk 3]
```diff
@@ -307,88 +277,110 @@ impl MockPeerManagerAdapter {
...
-        net_config,
-        vec![("test1", open_port())],
-        10,
-        routing_table_addr.clone(),
-    )
-    .0
-    .start();
-    (peer_manager_addr, routing_table_addr)
-}
+pub(crate) mod test_features {
```
Comment (matklad) [inline chain/network/src/test_utils.rs:281]: :+1: this looks much better. In general, I'd say that conditional imports are a smell, and that we should try to minimize those!

**Review by matklad — APPROVED** [review]

**Comment by pmnoxx** [comment]
@mm-near Can you take a look?

**Review by mm-near — APPROVED** [review]

**Pull request merged**